src/sema/statements.rs

```diff
@@ -331,6 +332,7 @@ fn statement(
                     ResolveTo::Type(&var_ty),
                 )?;
 
+                verify_result(&expr, ns, loc);
```

Comment:
You're evaluating the expression and checking the overflow, but you are not modifying the resulting AST.
If `expr` here is `1+2`, you evaluate it to `3`, but `expr` continues to be `1+2`. 
After evaluating, you can instead return the result to be used in the AST.